### PR TITLE
Update Multiplayer Automated Tests to use new Spawner

### PR DESCRIPTION
**Pull request by AMZN-Gene**
sv_defaultPlayerSpawnable no longer exists. Update tests to use new network player spawner tech.

**Important Note:** The tests fail run due to the fact that scripts attached to the spawnables have stopped working, but this is a good place to check-in as that is a different bug.

### CODE REVIEW

**Comment by AMZN-puvvadar** [comment]
This is quite a bit of duplication from MPSample. The thing with the spawner API is the implementation on the game side can be as complex or simple as you want. If you want/need RoundRobin for this project that's fine. You could also setup a cvar and spawn whatever thing you want at whatever position in AutomatedTestingSystemComponent::OnPlayerJoin.

**Comment by AMZN-Gene** [comment]
> This is quite a bit of duplication from MPSample. The thing with the spawner API is the implementation on the game side can be as complex or simple as you want. If you want/need RoundRobin for this project that's fine. You could also setup a cvar and spawn whatever thing you want at whatever position in AutomatedTestingSystemComponent::OnPlayerJoin.

RoundRobin is pretty great. If you have multiple clients spawning in (which some tests contain) it's nice so you can easily place spawn points in the level so they aren't all overlapping at <0,0,0>. I'm cool with RoundRobin being included in AutomatedTesting.

File: AutomatedTesting/Gem/Code/Source/AutomatedTestingSystemComponent.h [hunk 1]
```diff
@@ -39,5 +54,17 @@ namespace AutomatedTesting
         void Activate() override;
         void Deactivate() override;
         ////////////////////////////////////////////////////////////////////////
+
+        ////////////////////////////////////////////////////////////////////////
+        // Multiplayer::IMultiplayerSpawner interface implementation
+        Multiplayer::NetworkEntityHandle OnPlayerJoin(uint64_t userId, const Multiplayer::MultiplayerAgentDatum& agentDatum) override;
```
Comment (AMZN-puvvadar) [inline AutomatedTesting/Gem/Code/Source/AutomatedTestingSystemComponent.h:60]: NP: Format for these two functions is a but different. It'd be good to standardize.

**Review by AMZN-puvvadar — APPROVED** [review]
Minus the nitpick, looks good.

**Review by kberg-amzn — APPROVED** [review]
This is okay for now to get running, but after seeing this review I really want us to pursue the separate 'MultiplayerStarter' gem, or whatever we decide to call it, which contains all of these utility components factored out.

**Comment by kberg-amzn** [comment]
> > This is quite a bit of duplication from MPSample. The thing with the spawner API is the implementation on the game side can be as complex or simple as you want. If you want/need RoundRobin for this project that's fine. You could also setup a cvar and spawn whatever thing you want at whatever position in AutomatedTestingSystemComponent::OnPlayerJoin.
> 
> RoundRobin is pretty great. If you have multiple clients spawning in (which some tests contain) RoundRobin is nice in order to easily place spawn points in the level so players dont all spawn in and overlap at <0,0,0>. I'm cool with RoundRobin being included in AutomatedTesting.

Yeah, this is fine as long as the two implementations don't diverge prior to us factoring them out into a common gem.

File: AutomatedTesting/Gem/Code/Source/AutomatedTestingSystemComponent.h (hunk 1 above)
Comment (AMZN-Gene) [reply AutomatedTesting/Gem/Code/Source/AutomatedTestingSystemComponent.h:60]: Fixed!

**Pull request merged**